### PR TITLE
Introduce `assertResponseStatus` and `assertResponseData` helpers

### PR DESCRIPTION
**Pull request by danielbachhuber**

### CODE REVIEW

File: tests/class-wp-test-rest-testcase.php [hunk 1]
```diff
@@ -16,4 +16,21 @@ protected function assertErrorResponse( $code, $response, $status = null ) {
 			$this->assertEquals( $status, $data['status'] );
 		}
 	}
+
```
Comment (danielbachhuber) [inline tests/class-wp-test-rest-testcase.php:19]: @WP-API/amigos Could I get a second opinion on these?

Comment (rachelbaker) [inline tests/class-wp-test-rest-testcase.php:19]: @danielbachhuber LGTM

File: tests/test-rest-tags-controller.php [hunk 2]
```diff
@@ -305,7 +305,12 @@ public function test_get_items_search_args() {
 		$request = new WP_REST_Request( 'GET', '/wp/v2/tags' );
 		$request->set_param( 'search', 'App' );
 		$response = $this->server->dispatch( $request );
-		$this->assertEquals( 200, $response->get_status() );
+		$this->assertResponseStatus( 200, $response );
+		$this->assertResponseData( array(
+			array(
+				'name' => 'Apple',
+			),
+		), $response );
```
Comment (rmccue) [inline tests/test-rest-tags-controller.php]: Formatting here isn't right, I think. Newline before the first `array`, newline after `$response`

**Comment by joehoyle** [comment]
@danielbachhuber seeing quite a few errors in travis here, was it passing locally?

**Comment by danielbachhuber** [comment]
> seeing quite a few errors in travis here, was it passing locally?

I don't recall. I don't really want to follow through on this right now though.

**Pull request closed without merging**